hathor/indexes/utxo_index.py

```diff
@@ -0,0 +1,281 @@
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from abc import abstractmethod
+from dataclasses import dataclass
+from typing import Iterator, Optional
+
+from structlog import get_logger
+
+from hathor.conf import HathorSettings
+from hathor.indexes.base_index import BaseIndex
+from hathor.transaction import BaseTransaction, TxOutput
+from hathor.transaction.scripts import parse_address_script
+from hathor.util import sorted_merger
+
+logger = get_logger()
+settings = HathorSettings()
+
+
+@dataclass(frozen=True)
+class UtxoIndexItem:
+    token_uid: bytes
+    tx_id: bytes
+    index: int
+    address: str
+    amount: int
+    timelock: Optional[int]
+    heightlock: Optional[int]
+
+    def __repr__(self):
+        return (
+            'UtxoIndexItem('
+            f'token_uid={self.token_uid.hex()},'
+            f'tx_id={self.tx_id.hex()},'
+            f'index={self.index},'
+            f'address={self.address},'
+            f'amount={self.amount},'
+            f'timelock={self.timelock},'
+            f'heightlock={self.heightlock}'
+            ')'
+        )
+
+    @classmethod
+    def from_tx_output(cls, tx: BaseTransaction, index: int, tx_output: TxOutput) -> 'UtxoIndexItem':
+        assert tx.hash is not None
+
+        if tx_output.is_token_authority():
+            raise ValueError('UtxoIndexItem cannot be used with a token authority output')
+
+        address_script = parse_address_script(tx_output.script)
+        if address_script is None:
+            raise ValueError('UtxoIndexItem can only be used with scripts supported by `parse_address_script`')
+
+        tx_meta = tx.get_metadata()
+
+        heightlock: Optional[int] = tx_meta.height + settings.REWARD_SPEND_MIN_BLOCKS if tx.is_block else None
+        # XXX: timelock forced to None when there is a heightlock
+        timelock: Optional[int] = address_script.get_timelock() if heightlock is None else None
+        # XXX: that is, at least one of them must but None
+        assert timelock is None or heightlock is None
+
+        return cls(
+            token_uid=tx.get_token_uid(tx_output.get_token_index()),
+            tx_id=tx.hash,
+            index=index,
+            address=address_script.address,
+            amount=tx_output.value,
+            timelock=timelock,
+            heightlock=heightlock,
+        )
+
+
+def _should_skip_output(tx_output: TxOutput) -> bool:
+    if tx_output.is_token_authority():
+        return True
+    if parse_address_script(tx_output.script) is None:
+        return True
+    return False
+
+
+class UtxoIndex(BaseIndex):
+    """ Index of UTXOs by address+token
+
+    This index is currently optional and only used for an optional API.
+
+    It is expected to be called with only non-voided transactions and blocks, including the genesis. It ignores
```

Comment:
Yeah, I agree with you. We are handling correctly in the code, we are counting the created token output and ignoring the authority. My comment was just to update the docstring because we don't ignore new created tokens.

Reply:
Oh, got it! Will fix it.

Reply:
Done.